dot/services.go

```diff
@@ -90,18 +90,12 @@ func createStateService(cfg *Config) (*state.Service, error) {
 	return stateSrvc, nil
 }
 
-func createRuntime(cfg *Config, st *state.Service, ks *keystore.GlobalKeystore, net *network.Service) (runtime.Instance, error) {
+func createRuntime(cfg *Config, st *state.Service, ks *keystore.GlobalKeystore, net *network.Service, code []byte) (runtime.Instance, error) {
```

Comment:
you can load the code from `st.Storage.LoadCode()`

Reply:
I think then we need to pass the `block hash` as params. otherwise, we can only get the code corresponding to the best block.

Reply:
just realized, when the node restarts, do all the blocktree nodes get repopulated with the runtime? it doesn't look like the `blocktree.Encode` and `blocktree.Decode` functions have been updated